util/threadpool_imp.cc

```diff
@@ -155,6 +191,8 @@ void ThreadPoolImpl::Impl::JoinThreads(bool wait_for_jobs_to_complete) {
   // prevent threads from being recreated right after they're joined, in case
   // the user is concurrently submitting jobs.
   total_threads_limit_ = 0;
+  reserved_threads_ = 0;
```

Comment:
No need to take action - I just want to note that it's interesting we are setting reserved_threads_ = 0 while ThreadPoolImpl does not actually have the power to terminate those reserved threads. IMO, this is the biggest drawback of this approach exchanged for the benefit of implementational simplicity.